internal/helm/helm.go

```diff
@@ -109,6 +110,28 @@ func makeValues(app *types.App, targetDir string, env map[string]string, variabl
 	return ioutil.WriteFile(filepath.Join(targetDir, "values.yaml"), valuesRaw, 0644)
 }
 
+// convertTemplatesAny resolves templated fields in input, and returns marshaled YAML
+func convertTemplatesAny(input interface{}) ([]byte, error) {
+	stackData, err := yaml.Marshal(input)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to marshal stack data")
+	}
+	preStack := make(map[interface{}]interface{})
+	err = yaml.Unmarshal(stackData, preStack)
```

Comment:
I don't understand this marshal/unmarshal thing... Is it to cast input from `interface{}` to `map[interface{}]interface{}` ?

Reply:
Yes, input is an actual struct type, and we need a map[interface]interface to apply our convertTemplates

Reply:
What about this function https://github.com/docker/app/blob/master/types/settings/load.go#L84 ?
I think it's doing what you want?

Reply:
no, convertTemplates is replacing { template_foo: {value: bar}} into {foo: bar}, this is pretty unique to the helm generation.

Reply:
Right, but I was only refering to the marshal/unmarshal step...